core/modules/utils/filesystem.js

```diff
@@ -480,10 +480,12 @@ Delete a file described by the fileInfo if it exits
 */
 exports.deleteTiddlerFile = function(fileInfo,callback) {
 	//Only attempt to delete files that exist on disk
-	if(!fileInfo.filepath || !fs.existsSync(fileInfo.filepath)) {
+	if(!fileInfo || !fileInfo.filepath || !fs.existsSync(fileInfo.filepath)) {
 		//For some reason, the tiddler is only in memory or we can't modify the file at this path
-		$tw.syncer.displayError("Server deleteTiddlerFile task failed for filepath: "+fileInfo.filepath);
-		return callback(null,fileInfo);
+		var err = new Error("Invalid filepath `" + (fileInfo.filepath || "undefined") + "`");
```

Comment:
The error message seems to incorrectly assume that fileInfo is present. Maybe you need something like this:

   var err = new Error("Invalid filepath `" + (fileInfo && fileInfo.filepath ? fileInfo.filepath) : "undefined") + "`");